src/usdb_syncer/notes_parser.py

```diff
@@ -575,6 +576,30 @@ def minimum_song_length(self) -> str:
 
         return f"{minutes:02d}:{seconds:02d}"
 
+    def fix_relative_songs(self) -> None:
+        if not self.headers.relative:
+            return
+
+        offset = 0
+        for line in self.notes.all_lines():
+            if line.line_break and not line.line_break.in_time:
+                line.line_break.in_time = line.line_break.out_time
+
+            for note in line.notes:
+                note.start = note.start + offset  # type: ignore
+
+            if line.line_break and line.line_break.out_time and line.line_break.in_time:
+                line.line_break.out_time = line.line_break.out_time + offset  # type: ignore
+                line.line_break.in_time = line.line_break.in_time + offset  # type: ignore
+
+                offset = line.line_break.in_time
+
+                if line.line_break.out_time == line.line_break.in_time:
+                    line.line_break.in_time = None
```

Comment:
We shouldn't use `# type: ignore` too freely, because it could obscure real errors. In this case, there only seems to be a limitation with pylace and the comments are not required for our checks to pass, so I would rather not add them.
Anyway, I found a solution that is also understood by pylance. Please check that the logic is still correct.

Reply:
I fully agree, but tbh I did not understand why pylance complained operator `+` not being available for int and int | None, the latter referring to offset which is clearly initialized to 0 in line 583. So this was a temporary workaround to be able to submit the PR for further discussion.
I just checked, the code still works as expected. It is essential that relative songs are fixed before calling `maybe_split_duet_notes()`, otherwise this will fail on relative songs. So maybe `maybe_split_duet_notes() should double check that the song does not have relative timings.

Reply:
My best guess is that it was some kind of circular issue with `offset` and `in_time` setting each other to `int | None` due to the lines `offset = line.line_break.in_time` and `line.line_break.in_time = line.line_break.in_time + offset`.